database/rekordbox_pdb.ksy

```diff
@@ -391,33 +392,26 @@ types:
       - type: u1
         doc: |
           @flesniak says: "always 0x03, maybe an unindexed empty string"
-      - id: ofs_name
+      - id: ofs_name_near
         type: u1
         doc: |
           The location of the variable-length name string, relative to
-          the start of this row, unless subtype ix 0x64.
+          the start of this row, unless subtype is 0x64.
     instances:
+      ofs_name_far:
+        pos: _parent.row_base + 0x0a
+        type: u2
+        doc: |
+          For names that might be further than 0xff bytes from the
+          start of this row, this holds a two-byte offset, and is
+          signalled by the subtype value.
+        if: subtype == 0x64
       name:
+        pos: '_parent.row_base + (subtype == 0x64? ofs_name_far : ofs_name_near)'
```

Comment:
```python
ofs_name_far:
        pos: _parent.row_base + 0x0a
```

```
pos: '_parent.row_base + (subtype == 0x64? (_parent.row_base + 0x0a) : ofs_name_near)'

was it intended?

Reply:
Yes, that is really the weird layout that they have set up.

Reply:
Note that your substitution is not quite correct. In the case where it is a “far” string the offset from `_parent.row_base` is the two-byte value found at `_parent.row_base + 0x0a`. So row-base gets added to the value found there, it is not added twice to the resulting offset.